src/Database/PostgreSQL/PQTypes/Model/Trigger.hs

```diff
@@ -35,6 +38,38 @@ import Data.Text qualified as Text
 import Database.PostgreSQL.PQTypes
 import Database.PostgreSQL.PQTypes.SQL.Builder
 
+-- | Timing for a regault trigger.
+--
+-- @since 1.17.0.0
+data TriggerRegularTiming
+  = -- | An @AFTER@ trigger.
+    After
+  | -- | A @BEFORE@ trigger.
+    Before
+  deriving (Eq, Show)
+
+-- | Timing for a constraint trigger.
+--
+-- @since 1.17.0.0
+data TriggerConstraintTiming
```

Comment:
I would rename this to `ConstraintTiming`, since this is also applicable to e.g. foreign keys. Maybe in the future we can refactor the `ForeignKey` to use it.

Reply:
ok

Reply:
I had a look at the [Postgres grammar](https://github.com/postgres/postgres/blob/master/src/backend/parser/gram.y), in particular here [ConstraintAttr](https://github.com/postgres/postgres/blob/426ea611171da4e60ab4f3863fa3cc3683ae9547/src/backend/parser/gram.y#L4103) and here [ConstraintAttributeElem](https://github.com/postgres/postgres/blob/426ea611171da4e60ab4f3863fa3cc3683ae9547/src/backend/parser/gram.y#L6245) (this is the one relevant for triggers), and I think we can just use `ConstraintAttributes` :)